services/notifications.go

```diff
@@ -673,6 +661,13 @@ func queueEmailNotifications(notificationsByUserID map[uint64]map[types.EventNam
 				msg.Body += template.HTML(fmt.Sprintf("%s<br>====<br><br>", types.EventLabel[event_title]))
 				unsubURL := "https://" + utils.Config.Frontend.SiteDomain + "/notifications/unsubscribe"
 				for i, n := range ns {
+					// Find all unique notification titles for the subject
+					title := n.GetTitle()
+					if _, ok := notificationTitlesMap[title]; !ok {
+						notificationTitlesMap[title] = true
+						notificationTitles = append(notificationTitles, title)
+					}
```

Comment:
This triggers slightly different behavior than before:

Since the title for ETH Client updates is unique based on the client
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/f6f2c024da99729047f3724e67f7a7693d52ab06/services/notifications.go#L2111
this would now add 3 notifications to the mail's subject if 3 clients are updated.

Is this intended?

If not, maybe use n.GetEventName() to guarantee uniqueness but use the title within the subject. I did NOT test this thought though and honestly I think it's fine to add all 3 clients to the subject in this scenario as the subject gets shortened anyways.

Reply:
I compared this with the push notification implementation, see here
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/f6f2c024da99729047f3724e67f7a7693d52ab06/services/notifications.go#L517
where we have the same logic and just get every title:
```
for event, ns := range userNotifications {
	for _, n := range ns {
		added := false
		for _, userToken := range userTokens {
			notification := new(messaging.Notification)
			notification.Title = fmt.Sprintf("%s%s", getNetwork(), n.GetTitle())
		}
	}
}
```

Reply:
I think we pondered the same think. During the implementation I thought about the validator_is_offline EventName. It can also have multiple titles:
- Validator is Offline
- Validator Back Online
so your email subject could be "Validator is Offline and Validator Back Online" which made sense to me.